nextstrain/cli/util.py

```diff
@@ -339,6 +339,60 @@ def runner_name(runner: RunnerModule) -> str:
     return module_basename(runner).replace("_", "-")
 
 
+def runner_module(name: str) -> RunnerModule:
+    """
+    Converts a string *name* into a :py:cls:`RunnerModule`.
+
+    *name* is case-insensitive and underscores, hyphens, or spaces may be used
+    to separate words.  Internally, they're normalized to hyphens and the whole
+    string is lowercased.
+
+    ``native`` is accepted as an alias for ``ambient``.
+
+    Raises a :py:exc:`ValueError` if *name* is unknown.
+
+    >>> runner_module("docker") # doctest: +ELLIPSIS
+    <module 'cli.runner.docker' from '...'>
+
+    >>> runner_module("AWS-Batch") # doctest: +ELLIPSIS
+    <module 'cli.runner.aws_batch' from '...'>
+
+    >>> runner_module("AWS Batch") # doctest: +ELLIPSIS
+    <module 'cli.runner.aws_batch' from '...'>
+
+    >>> runner_module("native") # doctest: +ELLIPSIS
+    <module 'cli.runner.ambient' from '...'>
+
+    >>> runner_module("invalid")
+    Traceback (most recent call last):
+        ...
+    ValueError: invalid runtime name: 'invalid'; valid names are: 'docker', 'conda', 'ambient', 'aws-batch'
+
+    >>> runner_module("Invalid Name")
+    Traceback (most recent call last):
+        ...
+    ValueError: invalid runtime name: 'Invalid Name' (normalized to 'invalid-name'); valid names are: 'docker', 'conda', 'ambient', 'aws-batch'
+    """
+    # Import here to avoid circular import
+    from .runner import all_runners_by_name
+
+    normalized_name = re.sub(r'(_|\s+)', '-', name).lower()
+
+    # Accept "native" as an alias for "ambient" for backwards compatibility
+    if normalized_name == "native":
+        normalized_name = "ambient"
```

Comment:
Could we add a DeprecationWarning here? Mostly to inform existing `native` users of the new name, but also to justify eventual removal of the alias.

Reply:
I'd like not to. I'm not a fan of nagging users about things that still work just fine, and I wouldn't plan to remove `--native`. It is very little overhead to keep it around as an alias, we're unlikely to want to re-use it for something else in the future, and keeping it around will avoid breaking users' stuff.

If we do want to warn, we wouldn't use a DeprecationWarning, because [they're intended for developers and aren't visible by default to users](https://docs.python.org/3/library/exceptions.html#DeprecationWarning).